docs/Developing the HMI UI/index.md

```diff
@@ -0,0 +1,411 @@
+
+![Menu Icon](./assets/menu_circled.jpg)
+
+!!! NOTE
+If the user chooses to open the menu, the HMI must send a `UI.OnSystemContext` notification with the `SystemContext` enum: `MENU`. After the user exits the menu, another `UI.OnSystemContext` notification must be sent with the `SystemContext` value: `MAIN`.
+!!!
+
+The contents of the HMI's menu are populated by the RPC `UI.AddCommand`. Each `UI.AddCommand` received corresponds to an individual menu item. When the user selects a menu item, the HMI should send a `UI.OnCommand` notification. We believe it is best practice to exit the menu after a user makes a selection from the list of commands.
+
+There is some minor customization for the app menu. An HMI can choose to implement the app menu in a tile view, list view, or both. If an app has a preference for a type of menu layout, the HMI will receive a `UI.SetGlobalProperties` request from SDL Core.
+
+![List Menu](./assets/list_menu.png)
+
+![Tiles Menu](./assets/tiles_menu.png)
+
+## Implementing Popups
+
+There are several rpcs that are meant to display a popup or an overlay to the user.
+
+
+### `UI.Alert`
+
+Alert is a simple popup that can contain an image, text, and buttons.
+
+ ![Alert](./assets/alert.png)
+
+### `UI.PerformInteraction`
+
+`PerformInteraction` is a ppoup with contents displayed similar to the app menu
+
+![Perform Interaction](./assets/perform_interaction.png)
+
+### `UI.Slider`
+
+`Slider` is a popup that allows the user to enter an input via a slider button.
+
+![Slider](./assets/slider.png)
+
+### `UI.ScrollableMessage`
+
+![Scrollable Message](./assets/scrollable_message.png)
+
+`ScrollableMessage` is a popup that allows the user to read a longer message that requires scrolling.
+
+!!! NOTE
+
+It is important that the HMI sends SDL Core a `UI.OnSystemContext` notification when displaying and then hiding a popup. `UI.Alert` is the only popup to have it's own SystemContext enum `ALERT`. All other popups must use the SystemContext enum value, `HMI_OBSCURED`.
+
+!!!
+
+## Navigating Through the IVI
+
+It is common for an SDL UI to be integrated into an existing OEM's UI. In order for SDL Core to work well with a head unit that has other embedded components, the HMI should make use of the `BasicCommunication.OnEventChanged` notification. This notification allows connected SDL applications to receive updates about their HMI statuses when a user interacts with other components like the embedded navigation or radio. For example if an SDL media application has been activated and is playing audio, and a user selects to switch the audio source to the embedded radio, the HMI should send SDL Core a `BasicCommunication.OnEventChanged` notification with `eventName = AUDIO_SOURCE` and `isActive = true`. This HMI notification will let the media application know that it no longer has control of the audio source and its SDL capabilities are overwritten by the embedded audio's capabilities.
```

Comment:
```suggestion
It is common for an SDL UI to be integrated into an existing OEM's UI. In order for SDL Core to work well with a head unit that has other embedded components, the HMI should make use of the `BasicCommunication.OnEventChanged` notification. This notification allows connected SDL applications to receive updates about their HMI status when a user interacts with other components like the embedded navigation or radio. For example, if an SDL media application is active and is playing audio, then the user switches the audio source to the embedded radio, the HMI should send SDL Core a `BasicCommunication.OnEventChanged` notification with `eventName = AUDIO_SOURCE` and `isActive = true`. This HMI notification will let the media application know that it no longer has control of the audio source.
```
Not certain what to replace `and its SDL capabilities are overwritten by the embedded audio's capabilities` with, but I don't feel the wording is quite clear there.